linode/manager
# feat: [M3-7645] – Support VPC IPv4 Ranges in Add/Edit Linode Config dialog

## Description 📝
Add support for VPC IPv4 ranges in Add/Edit Linode Config dialog

## Changes  🔄
- Remove some `vpcEnabled`/`isFeatureEnabled` logic in `LinodeConfigDialog.tsx`
- Logic to include `ip_ranges` in config payload, display individual IP range errors

## Preview 📷
<details>
<summary>Starting state, before any ranges have been assigned</summary>

![Screenshot 2024-02-09 at 1 41 29 PM](https://github.com/linode/manager/assets/114682940/25ea8862-4e2b-4812-a04a-2d5e656c83da)
</details>

## How to test 🧪
### Verification steps 
For a Linode already assigned to a VPC, go to Linode Details > Configurations > Edit and confirm that the info displayed for the VPC interface is correct (particularly the IP ranges -- existing IP ranges should be displayed, but if there are none, you should just see the "Assign additional IPv4 ranges" section with no input fields to start).

Try adding and deleting IP ranges and ensure the payload (PUT request to `linode/instances/{linode_id}/configs/{config_id}`) and UI matches what you'd expect.

Note: The CIDR notation (e.g., `/24`) of the range needs to be narrower (i.e., a larger number) than the subnet IP range. For example, if the subnet IP range has `/16`, the CIDR notation for the ranges could be `/24`.

## As an Author I have considered 🤔
